src/services/offer.js

```diff
@@ -262,6 +228,59 @@ class OfferService {
             { results };
     }
 
+    /**
+     * Builds a search query with a regular find().
```

Comment:
Same comment for this documentation. Explain what the function does and when to use, not how it does it.

Maybe it would be valuable to name it something like `_buildInitialSearchQuery` to more clearly distinguish between the one used on the first search, and the subsequent ones